olive/passes/pytorch/merge_lora_weights.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+from typing import Any, Dict
+
+import torch
+
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import PyTorchModelHandler
+from olive.passes import Pass
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class MergeLoraWeights(Pass):
+    """Merge Lora weights into a complete model."""
+
```

Comment:
- Please use "adapter" instead of "lora" in naming. Say MergeAdapterWeights. LoRA is too specific.
- Does this merge the adapter weights into master weights OR add rank adaption (code + weight) into the model?

Reply:
Updated. 
Also for ` adapter weights into master weights` and `rank adaption (code + weight)`, could you help explain the differences? 
From my understanding, current pass will merge the adapter weights into base model, at the time it also unload the rank adaption code from base model. So I guess the answer is `adapter weights into master weights`.
https://huggingface.co/docs/peft/v0.6.2/en/package_reference/tuners#peft.LoraModel.merge_and_unload

Reply:
I see. Thanks!